moveit_planners/ompl/CMakeLists.txt

```diff
@@ -1,10 +1,41 @@
 cmake_minimum_required(VERSION 3.22)
 project(moveit_planners_ompl LANGUAGES CXX)
 
+if(APPLE)
+  # Manually tell CMake how to find OpenMP
+  set(OpenMP_C_FLAGS "-Xclang -fopenmp")
+  set(OpenMP_CXX_FLAGS "-Xclang -fopenmp")
+  set(OpenMP_C_LIB_NAMES "omp")
+  set(OpenMP_CXX_LIB_NAMES "omp")
+  set(OpenMP_omp_LIBRARY "/opt/homebrew/opt/libomp/lib/libomp.dylib")
```

Comment:
Please do not hardcode paths or options on Apple. This would break compilation when openmp is installed in any other package manager that is not homebrew on Apple Silicon, see https://github.com/SteveMacenski/slam_toolbox/pull/793#issuecomment-3090261194 for more detials. All those variables can configured externally, and if that is not possible please add CMake options to set the settings you need, but do not hardcoded them on Apple to avoid breaking existing users.